<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
  <head>
    <meta content="text/html; charset=UTF-8" http-equiv="content-type">
    <title>gene2path:programs</title>
  </head>
  <body>
    <div style="text-align: center;"><big><big style="font-weight: bold;"></big></big>
      <hr>
      <div align="left"><b>Programs</b><big><big style="font-weight: bold;"> </big></big></div>
      <hr><big><big style="font-weight: bold;"><br>
        </big></big>
      <div style="text-align: justify;"><a href="dictionary.cgi">Dictionary</a><br>
        The sequences for a selected group of genes (list/Accession/Number) can
        be obtained in a “fasta” format using the getFASTA program. The gene
        list contains the “Accession Numbers” from GenBank<span style="background: #ffff00"></span>.
        <br>
        <br>
        <a href="microarray.cgi">Microarray analysis</a><br>
        The programme automates the analysis of the data generated by the
        microarray scanner. In our case, values from the "g" signal process file
        in ".txt" format were used. Among the programme's options, the user can
        choose whether the data files correspond to control or experimental
        groups, and the level of gene expression to be visualized on the web
        site can be established. <br>
        <br>
        <a href="homology.cgi">Orthology</a><br>
        <span lang="en-GB">By searching three NCBI databases, the "Orthology"
          program provides a tool to identify a homologous gene in another
          species. This program requires only two parameters to be entered: the
          "accession number" corresponding to the identification code of the
          gene selected; and the "species" for which the search is being
          performed.</span> <br>
        <br>
        <a href="pathways.cgi">Pathways</a><br>
        <span lang="en-GB">To obtain KEGG lists for the gene identified in each
          species (according to the gene ID). Thus, the program searches for the
          respective gene from a given species in the KEGG_PATHWAY database (</span><font
          color="#000080"><span lang="zxx"><u><a class="western" href="http://www.genome.jp/kegg/pathway.html">http://www.genome.jp/kegg/pathway.html</a></u></span></font><span
          lang="en-GB">), obtaining and examining the KEGG</span><span lang="en-GB"><span
            style="background: #ffff00"></span></span><span lang="en-GB">&nbsp;</span><font
          style="font-size: 8pt" size="1"></font><span lang="en-GB">lists
          (keggData) to define their ID</span><span lang="en-GB"><span style="background: #ffff00"></span></span><span
          lang="en-GB">, url, description name and additional data.</span>
        <p class="western" style="margin-bottom: 0in"> </p>
        <a href="inverse.cgi">Inverse Search</a><br>
        <span lang="en-GB">This step is that of finding all the genes involved
          in a particular KEGG pathway that are identified by a keggID label.
          All the data related to the selected KEGG are obtained from the KEGG2
          database (</span><font color="#000080"><span lang="zxx"><u><a class="western"
                href="http://www.genome.jp/kegg/kegg2.html">http://www.genome.jp/kegg/kegg2.html</a></u></span></font><span
          lang="en-GB">) and the data obtained are saved in a file that
          contains: the KEGGData image, the module information for which the
          conversion codes of the image-gene number are stored; and the
          respective sequences in FASTA format.</span> <br>
      </div>
    </div>
  </body>
</html>
